java/flight/flight-core/src/test/java/org/apache/arrow/flight/auth2/TestBasicAuth2.java

```diff
@@ -99,23 +98,26 @@ public void getStream(CallContext context, Ticket ticket, ServerStreamListener l
     };
   }
 
-  private void startServerAndClient() throws IOException {
+  private static void startServerAndClient() throws IOException {
     final FlightProducer flightProducer = getFlightProducer();
-    this.server = FlightServer
+    server = FlightServer
         .builder(allocator, forGrpcInsecure(LOCALHOST, 0), flightProducer)
         .headerAuthenticator(new GeneratedBearerTokenAuthenticator(
-            new BasicCallHeaderAuthenticator(this::validate)))
+            new BasicCallHeaderAuthenticator(TestBasicAuth2::validate)))
         .build().start();
-    this.client = FlightClient.builder(allocator, server.getLocation())
+    client = FlightClient.builder(allocator, server.getLocation())
         .build();
   }
 
-  @AfterEach
-  public void shutdown() throws Exception {
-    AutoCloseables.close(client, client2, server, allocator);
+  @AfterAll
+  public static void shutdown() throws Exception {
+    AutoCloseables.close(client, client2, server);
     client = null;
     client2 = null;
     server = null;
+
+    allocator.getChildAllocators().forEach(BufferAllocator::close);
+    AutoCloseables.close(allocator);
     allocator = null;
```

Comment:
These tests were flakey before adding this, so it's likely they do correctly close() but there's a race condition that causes them to get missed. I experimented (changed these tests to use a separate allocator for FlightClient and FlightServer) and found that it's the Flight server's allocator that leaks. I _think_ it relates to the server taking too long to shutdown during FlightServer.close(), which mentions it will leak resources if shutdown is taking too long.

I'll create a new Issue.

Reply:
Created #38735 for this.

Reply:
It's definitely not a design choice. And the only use of child allocators I see in flight is in tests, except for one created in the FlightClient (which is closed).